### ISSUE TITLE
Fetch does not await of the response listener

### ISSUE THREAD

**Issue by mikicho** [mswjs/interceptors#656]
Following [this change](https://github.com/mswjs/interceptors/pull/591), we need to update the behavior for `fetch` as well.

```js
const { FetchInterceptor } = require('@mswjs/interceptors/fetch')

const interceptor = new FetchInterceptor({
  name: 'my-interceptor',
})
interceptor.apply();
interceptor.on('response', async ({ request }) => {
  await new Promise(r => setTimeout(r, 100));
  console.log(2);
});

(async function () {
  console.log(1);
  await fetch('http://example.com')
  console.log(3);
})()
```
Actual:
>1
3
2

Expected:
> 1
2
3

@kettanaito I believe I can make this change, but I want to ensure that there are no unintended side effects that I haven't considered with this change.

**Comment by kettanaito** [comment]
Hi, @mikicho. That actual order doesn't seem right. Go ahead to open a pull request to fix this 👍 I will take a look how we are handling this in MockHttpSocket during the review to make sure we stay consistent.

**Comment by kettanaito** [comment]
## Released: v0.36.4 🎉

This has been released in v0.36.4!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.36.4)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.36.4)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
